chain/consensus/filcns/upgrades.go

```diff
@@ -2468,6 +2480,108 @@ func upgradeActorsV13VerifregFix(oldBuggyVerifregCID, newManifestCID cid.Cid) fu
 	}
 }
 
+func PreUpgradeActorsV14(ctx context.Context, sm *stmgr.StateManager, cache stmgr.MigrationCache, root cid.Cid, epoch abi.ChainEpoch, ts *types.TipSet) error {
+	// Use half the CPUs for pre-migration, but leave at least 3.
+	workerCount := MigrationMaxWorkerCount
```

Comment:
This is more of a meta point but it looks like all of this parameter preparation can be extracted to a function and called for all future upgrades.  Idk if this code is still lying around for other upgrade functions but if so it could reduce the copy paste volume.

Reply:
Thanks for poiting this out. I have opened a [issue](https://github.com/filecoin-project/lotus/issues/11971), and Aarsh, me and Rod will look into this once we start the `Ease Lotus Maintenance and Development` workstream